### PR TITLE
feat(DIGITAL-2947): Conditional pages

### PR DESCRIPTION
**Pull request by lydiagray**
### Description
Updated FormSchema to have RenderConditions on pages, added validation for page RenderConditions which uses the new Conditions Validation, moved RemoveSession, amended and added new unit tests. Updated wiki

### Checklist
- [X ] Code compiles correctly
- [X ] Created tests for the new changes
- [X ] All tests passing
- [X ] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/Models/Page.cs [hunk 1]
```diff
@@ -62,21 +66,21 @@ public IEnumerable<BaseProperty> InvalidElements
...
+            element.Type == EElementType.Checkbox ||
+            element.Type == EElementType.Declaration ||
+            element.Type == EElementType.Address ||
+            element.Type == EElementType.AddressManual ||
+            element.Type == EElementType.DateInput ||
+            element.Type == EElementType.TimeInput ||
+            element.Type == EElementType.DatePicker ||
+            element.Type == EElementType.Street ||
+            element.Type == EElementType.Organisation ||
+            element.Type == EElementType.FileUpload
```
Comment (lallysmbc) [inline src/Models/Page.cs:83]: EElementType.Map is missing from check

File: src/Models/FormSchema.cs [hunk 2]
```diff
@@ -38,19 +38,20 @@ public FormSchema()
...
         {
-            Page page;
             try
             {
-                page = Pages.SingleOrDefault(_ => _.PageSlug.ToLower().Trim() == path.ToLower().Trim());
+                var pages = Pages.Where(_ => _.PageSlug.ToLower().Trim() == path.ToLower().Trim()).OrderByDescending(_ => _.RenderConditions.Count).ToList();
+
+                var page = pageHelper.GetPageWithMatchingRenderConditions(pages);
+
+                return page;
```
Comment (lallysmbc) [inline src/Models/FormSchema.cs:49]: if there is only a single page, can we return straight away as there is no need to process any conditions also will reduce calls to the cache

File: src/Models/Page.cs (hunk 1 above)
Comment (lallysmbc) [reply src/Models/Page.cs:83]: ignore this, map is not in the ValidatableElements.

File: src/Models/FormSchema.cs (hunk 2 above)
Comment (lydiagray) [reply src/Models/FormSchema.cs:49]: Good point, will fix shortly

Comment (lydiagray) [reply src/Models/FormSchema.cs]: Fix pushed

**Pull request merged**